File: src/main/java/com/zoarial/jpa/converters/PathConverter.java
package com.zoarial.jpa.converters;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.nio.file.Path;
import java.nio.file.Paths;

@Converter
public class PathConverter implements AttributeConverter<Path, String> {

    @Override
    public String convertToDatabaseColumn(Path path) {
        return path.toFile().getAbsolutePath();
    }

    @Override
    public Path convertToEntityAttribute(String s) {
        return Paths.get(s);
    }
}

File: src/main/java/com/zoarial/jpa/converters/PathConverter.kt
package com.zoarial.jpa.converters

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.nio.file.Path
import java.nio.file.Paths

@Converter
class PathConverter : AttributeConverter<Path, String> {
    override fun convertToDatabaseColumn(path: Path): String {
        return path.toFile().absolutePath
    }

    override fun convertToEntityAttribute(s: String): Path {
        return Paths.get(s)
    }
}